src/scopetools.js

```diff
@@ -101,6 +101,10 @@ function createScopes(node, parent) {
 
     } else if (node.type === "CatchClause") {
         const identifier = node.param;
+        
+        if (identifier === null) {
```

Comment:
The `node.$scope = new Scope(...)` part should still happen. I suggest we make the code look like this:

```js
else if (node.type === "CatchClause") {
    node.$scope = new Scope({
        kind: "catch-block",
        node: node,
        parent: parent.$scope,
    });

    const identifier = node.param;
    if (identifier !== null) {
        // Rest of the code here
    }
}
```